tools/fuzzer/expr_gen.cc

```diff
@@ -1381,7 +1386,7 @@ Enum weighted_pick(
   RealType running_sum = 0;
   for (size_t i = 0; i < array.size(); i++) {
     running_sum += mask[i] ? array[i] : 0;
-    if (choice < running_sum) {
+    if (choice <= running_sum) {
```

Comment:
Note: there's a slight possibility that RNG generates `choice` equal to the total sum, in which none of the options will be selected, resulting in assertion error.